starter-code/src/App.js

```diff
@@ -1,18 +1,76 @@
 import React, { Component } from 'react';
-import logo from './logo.svg';
-import './App.css';
+import 'bulma/css/bulma.css';
+import Foods from './datasets/foods.json';
+import Header from './components/misc/Header';
+import FoodBox from './components/FoodBox';
+import Search from './components/Search';
+import Menu from './components/Menu';
+import AddFoodForm from './components/AddFoodForm.js';
 
 class App extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      foods: [...Foods],
+      menu: []
+    }
+  }
+
+  onFilter = (search) => {
+    const newFoods = Foods.filter(food => {
+      return food.name.toLowerCase().includes(search.toLowerCase());
+    });
+    this.setState({foods: newFoods});
+  }
+
+  onModifyMenu = (food) => {
+    let oldMenu = this.state.menu;
+
+    const itemModified = oldMenu.find((item) => item.name === food.name );
+    if ( itemModified ) { 
+      food.quantity = parseInt(food.quantity, 0) + parseInt(itemModified.quantity, 0); 
+      oldMenu = oldMenu.filter((item) => item.name !== food.name );
+    };
+    const newMenu = [...oldMenu, food];
+    this.setState({menu: newMenu})
+  }
+
+  onDeleteMenu = (food) => {
+    const newMenu = this.state.menu.filter((item) => item.name !== food.name );;
+    this.setState({menu: newMenu})
+  }
+
+  onAddFood = (food) => {
+    const foodsCloned = [...this.state.foods];
+    foodsCloned.push(food);
+    this.setState({
+      foods: foodsCloned
+    });
```

Comment:
Ese `push` me da picores, usa `concat()` o `spread operator`. Podrías resolverlo mejor de la siguiente manera: `this.setState({ foods: [...this.state.foods, food] })`. Esto hace justo lo mismo.